app/services/imports/lettings_logs_import_service.rb

```diff
@@ -77,13 +77,15 @@ def create_log(xml_doc)
       attributes["rent_type"] = rent_type(xml_doc, attributes["lar"], attributes["irproduct"])
       attributes["hhmemb"] = household_members(xml_doc, previous_status)
       (1..8).each do |index|
-        attributes["age#{index}"] = safe_string_as_integer(xml_doc, "P#{index}Age")
-        attributes["age#{index}_known"] = age_known(xml_doc, index, attributes["hhmemb"])
-        attributes["sex#{index}"] = sex(xml_doc, index)
-        attributes["ecstat#{index}"] = unsafe_string_as_integer(xml_doc, "P#{index}Eco")
+        person_index = people_with_details_ids(xml_doc)[index - 1] || index
```

Comment:
won't this `|| index` condition mean e.g. if we have details for people 1 and 3 given, then we return person 3's xml details for both the second AND third pass of this loop?

Reply:
This is a very good point, updated it now and changed the test a bit as well. Thanks!